src/Core/Extensible.php

```diff
@@ -525,7 +530,16 @@ public function getExtensionInstance($extension)
     public function hasExtension($extension)
     {
         $instances = $this->getExtensionInstances();
-        return isset($instances[$extension]);
+        if (array_key_exists($extension, $instances)) {
+            return true;
+        }
+        // Injector can be used to override the class used
+        foreach ($instances as $instance) {
+            if (is_a($instance, $extension)) {
+                return true;
+            }
+        }
+        return false;
```

Comment:
Can we check by reusing `getExtensionInstances()` here instead of duplicating its logic? It doesn't look like it would be any extra overhead

Reply:
Good call - done